File: backend/app/schemas/badge.py
from pydantic import BaseModel

from app.schemas.user import UserBriefResponse


class BadgeResponse(BaseModel):
    slug: str
    name: str
    description: str
    earned: bool


class BadgeListResponse(BaseModel):
    badges: list[BadgeResponse]


class ContributorResponse(BaseModel):
    rank: int
    user: UserBriefResponse
    vote_count: int


class ContributorRankingResponse(BaseModel):
    contributors: list[ContributorResponse]
    period: str

File: frontend/src/types/badge.ts
import type { UserBrief } from "./video";

export interface Badge {
  key: string;
  label: string;
  description: string;
  icon: string;
  earned: boolean;
}

export interface Contributor {
  rank: number;
  user: UserBrief;
  vote_count: number;
}
